packages/angular_devkit/build_angular/src/webpack/plugins/service-worker-plugin.ts

```diff
@@ -31,17 +37,25 @@ export class ServiceWorkerPlugin {
         throw new Error('Compilation output path cannot be empty.');
       }
 
-      await augmentAppWithServiceWorker(
-        projectRoot,
-        root,
-        outputPath,
-        baseHref,
-        ngswConfigPath,
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (compiler.inputFileSystem as any).promises,
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        (compiler.outputFileSystem as any).promises,
-      );
+      try {
+        await augmentAppWithServiceWorker(
+          projectRoot,
+          root,
+          outputPath,
+          baseHref,
+          ngswConfigPath,
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (compiler.inputFileSystem as any).promises,
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (compiler.outputFileSystem as any).promises,
+        );
+      } catch (error) {
+        compilation.errors.push(
+          new WebpackError(
```

Comment:
```suggestion
          new compilation.compiler.webpack.WebpackError(
```
Removes the need to change the import and also guarantees the same Webpack instance that is executing the plugin.